app/services/billable_metrics/aggregations/sum_service.rb

```diff
@@ -54,7 +56,55 @@ def compute_pay_in_advance_aggregation
         return BigDecimal(0) unless event
         return BigDecimal(0) if event.properties.blank?
 
-        BigDecimal(event.properties.fetch(billable_metric.field_name, 0).to_s)
+        unless previous_event
+          value = event.properties.fetch(billable_metric.field_name, 0).to_s
+          update_event_metadata(current_aggregation: value, max_aggregation: value)
+
+          return BigDecimal(value)
+        end
+
+        current_aggregation = BigDecimal(previous_event.metadata['current_aggregation']) +
+                              BigDecimal(event.properties.fetch(billable_metric.field_name, 0).to_s)
+
+        old_max = BigDecimal(previous_event.metadata['max_aggregation'])
+
+        result = if current_aggregation > old_max
+          update_event_metadata(current_aggregation:, max_aggregation: current_aggregation)
+
+          current_aggregation - old_max
+        else
+          update_event_metadata(current_aggregation:, max_aggregation: old_max)
+
+          0
+        end
+
+        BigDecimal(result)
+      end
+
+      private
+
+      def date_service
+        @date_service ||= Subscriptions::DatesService.new_instance(
+          subscription,
+          Time.current,
```

Comment:
We should not use `Time.current` at this level as it will prevent us from replaying the aggregation with a past date (for example to retry a failed job, of refreshing an invoice).

Do you think we could get this value injected into the aggregator at init?

Reply:
added `event.timestamp`